source/command-logging-and-monitoring/command-logging-and-monitoring.rst

```diff
@@ -0,0 +1,615 @@
+     */
+    duration: Int64;
+
+    /**
+     * Returns the command name.
+     */
+    commandName: String;
+
+    /**
+     * Returns the failure. Based on the language, this SHOULD be a message string, exception
+     * object, or error document.
+     */
+    failure: String,Exception,Document;
+
+    /**
+     * Returns the client generated request id.
+     */
+    requestId: Int64;
+
+    /**
+     * Returns the driver generated operation id. This is used to link events together such
+     * as bulk write operations. OPTIONAL.
+     */
+    operationId: Int64;
+
+    /**
+     * Returns the connection id for the command. For languages that do not have this,
+     * this MUST return the driver equivalent which MUST include the server address and port.
+     * The name of this field is flexible to match the object that is returned from the driver.
+     */
+    connectionId: ConnectionId;
+
+    /**
+     * Returns the server connection id for the command. The server connection id is distinct from
+     * the connection id and is returned by the hello or legacy hello response as "connectionId"
+     * from the server on 4.2+. Drivers MAY use a wider type to represent the server connection ID
+     * value, but the server's behavior is to return an Int32.
+     */
+    serverConnectionId: Optional<Int32>;
+
+    /**
+     * Returns the service id for the command when the driver is in load balancer mode.
+     * For drivers that wish to include this in their ConnectionId object, this field is
+     * optional.
+     */
+    serviceId: Optional<ObjectId>;
+  }
+
+------------
+Log Messages
+------------
+Please refer to the `logging specification <../logging/logging.rst>`_ for details on logging implementations in general, including log levels, log
+components, and structured versus unstructured logging.
+
+Drivers MUST support logging of command information via the following types of log messages. These messages MUST be logged at ``Debug`` level and use
+the ``command`` log component.
+
+The log messages are intended to match the information contained in the events above. Drivers MAY implement command logging
+support via an event subscriber if it is convenient to do so.
+
+The types used in the structured message definitions below are demonstrative, and drivers MAY use similar types instead so long as the information
+is present (e.g. a double instead of an integer, or a string instead of an integer if the structured logging framework does not support numeric types.) 
+
+For the string representations of messages defined below, for any optional field that is not available, drivers MAY either substitute the value with a
+language-appropriate representation of null (e.g. "null", "nil", "None") or omit the corresponding segment of the message altogether. 
+
+Drivers MUST not emit command log messages for commands issued as part of the handshake with the server, or heartbeat commands issued by server monitors. 
```

Comment:
Rust doesn't emit command monitoring events for these commands already so this was easy for us and consistent with the monitoring APIs. but not sure if the same is true for other drivers?

Reply:
It's true for Java driver

Reply:
IIRC libmongoc has trace-level logging that can dump all data sent over sockets. This includes messages pertaining to handshakes, auth, and monitoring. /cc @kevinAlbs

Reply:
Same for .Net. But in my implementation logs for heartbeats commands were emitted. It was an easy fix, but wondering whether this can/worth to be tested .

Reply:
I added some tests, `no-heartbeat-messages.yml` and `no-handshake-messages.yml`, that I think at least implicitly test this. 
the heartbeat one creates a client, waits for a server description change (implying a heartbeat happened), and then runs a ping, and then asserts we have only seen command logs for the ping.
the handshake one creates a client, runs a ping, waits for connection created and ready events (implying a handshake happened), then runs a ping, and then asserts we have only seen command logs for the ping.

let me know what you think and if those pass for you! cc @jyemin could you take a look and try in Java too?